static/app/views/issueList/overviewFc.tsx

```diff
@@ -0,0 +1,1237 @@
+  const [queryCount, setQueryCount] = useState(0);
+  const [queryCounts, setQueryCounts] = useState<QueryCounts>({});
+  const [queryMaxCount, setQueryMaxCount] = useState(0);
+  const [error, setError] = useState<string | null>(null);
+  const [issuesLoading, setIssuesLoading] = useState(true);
+  const [memberList, setMemberList] = useState<ReturnType<typeof indexMembersByProject>>(
+    {}
+  );
+  const undoRef = useRef(false);
+  const lastRequestRef = useRef<Request | null>(null);
+  const lastStatsRequestRef = useRef<Request | null>(null);
+  const lastFetchCountsRequestRef = useRef<Request | null>(null);
+  const pollerRef = useRef<CursorPoller | undefined>(undefined);
+  const actionTakenRef = useRef(false);
+
+  const groups = useLegacyStore(GroupStore);
+  useEffect(() => {
+    const storeGroupIds = groups.map(group => group.id).slice(0, MAX_ISSUES_COUNT);
+    if (!isEqual(groupIds, storeGroupIds)) {
+      setGroupIds(storeGroupIds);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [groups]);
+
+  useIssuesINPObserver();
+
+  const onRealtimePoll = useCallback(
+    (data: any, {queryCount: newQueryCount}: {queryCount: number}) => {
+      // Note: We do not update state with cursors from polling,
+      // `CursorPoller` updates itself with new cursors
+      GroupStore.addToFront(data);
+      setQueryCount(newQueryCount);
+    },
+    []
+  );
+
+  useEffect(() => {
+    pollerRef.current = new CursorPoller({
+      linkPreviousHref: parseLinkHeader(pageLinks)?.previous?.href,
+      success: onRealtimePoll,
+    });
+  }, [onRealtimePoll, pageLinks]);
+
+  const getQueryFromSavedSearchOrLocation = useCallback(
+    (props: Pick<Props, 'savedSearch' | 'location'>): string => {
+      if (
+        !organization.features.includes('issue-stream-custom-views') &&
+        props.savedSearch
+      ) {
+        return props.savedSearch.query;
+      }
+
+      const {query} = props.location.query;
+
+      if (query !== undefined) {
+        return decodeScalar(query, '');
+      }
+
+      return DEFAULT_QUERY;
+    },
+    [organization.features]
+  );
+
+  const getSortFromSavedSearchOrLocation = useCallback(
+    (props: Pick<Props, 'savedSearch' | 'location'>): string => {
+      if (!props.location.query.sort && props.savedSearch?.id) {
+        return props.savedSearch.sort;
+      }
+
+      if (props.location.query.sort) {
+        return props.location.query.sort as string;
+      }
+      return DEFAULT_ISSUE_STREAM_SORT;
+    },
+    []
+  );
+
+  const getQuery = useCallback((): string => {
+    return getQueryFromSavedSearchOrLocation({
+      savedSearch,
+      location,
+    });
+  }, [getQueryFromSavedSearchOrLocation, savedSearch, location]);
+
+  const getSort = useCallback((): string => {
+    return getSortFromSavedSearchOrLocation({
+      savedSearch,
+      location,
+    });
+  }, [getSortFromSavedSearchOrLocation, savedSearch, location]);
```

Comment:
is useMemo not the same here. i think location will update the callback every time

Reply:
It sure is - just a result of me copying 1 for 1 from the old component. Converted to simple values